File: plugins/harness/skills/harness-install/assets/maven/harness-maven-plugin/src/main/java/com/ririnto/sinon/harness/Finding.java
package com.ririnto.sinon.harness;

/**
 * Record representing a validation finding with severity, category, message, and
 * optional structured location and fix metadata. Line and column values are
 * 1-indexed.
 */
public record Finding(
        String severity,
        String category,
        String message,
        String file,
        Integer startLine,
        Integer startColumn,
        Integer endLine,
        Integer endColumn,
        FindingFix fix) {
    /**
     * Convenience factory for the legacy 3-argument shape.
     *
     * @param severity severity level
     * @param category manifest category key
     * @param message  human-readable violation message
     * @return finding with no location or fix metadata
     */
    public static Finding of(String severity, String category, String message) {
        return new Finding(severity, category, message, null, null, null, null, null, null);
    }
}

File: plugins/harness/skills/harness-install/templates/gradle/docs/harness/gradle-plugin/src/main/kotlin/ai/harness/gradle/Finding.kt
package ai.harness.gradle

/**
 * Represents the severity level of a harness validation finding.
 */
enum class Severity {
	ERROR,
	WARN,
	INFO,
}

/**
 * Represents a validation finding with severity, category, and message.
 */
data class Finding(
	val severity: Severity,
	val category: String,
	val message: String,
)
